npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -51,12 +51,29 @@ def self.yarn_berry?(yarn_lock)
       end
 
       def self.yarn_major_version
-        @yarn_major_version ||= fetch_yarn_major_version
-      end
-
-      def self.fetch_yarn_major_version
+        retries = T.let(0, Integer)
         output = SharedHelpers.run_shell_command("yarn --version")
         Version.new(output).major
+      rescue Dependabot::SharedHelpers::HelperSubprocessFailed => e
+        # Should never happen, can probably be removed once this settles
+        raise "Failed to replace ENV, not sure why" if retries.positive?
```

Comment:
One reason might be if there are multiple env vars in the file? Or potentially multiple files with envs?

Could we do a global cleanup of variables in the file to get around the 1st issue?

Reply:
My idea was to let this error show up in Sentry and use it to learn about the edge cases. But I did push the fix you suggested to the first edge case!